### PR TITLE
fix(mojaloop/#2439): quoting-service-model.validatequoterequest-doesnt-perform-correct-validation

### PR DESCRIPTION
**Pull request by mdebarros**
fix([mojaloop/#2439](https://github.com/mojaloop/project/issues/2439)): quoting-service model.validateQuoteRequest doesn't perform correct validation when simpleRoutingMode is TRUE
- added typesafe checks for validate quote request logic
- added devspace patterns to gitignore
- minor formatting of the serverStart unit tests for clarity
- updated dependencies to latest version
- added circleci config for automated releases
- added standard-version dependency for automated releases
- fixed audit resolve issues:
```text
--------------------------------------------------
 tar needs your attention.

[ high ] Arbitrary File Creation/Overwrite via insufficient symlink protection due to directory cache poisoning using symbolic links
 vulnerable versions <4.4.18 || >=5.0.0 <5.0.10 || >=6.0.0 <6.1.9 found in:
 - dependencies: @mojaloop/event-sdk>grpc>@mapbox/node-pre-gyp>tar
[ high ] Arbitrary File Creation/Overwrite on Windows via insufficient relative path sanitization
 vulnerable versions <4.4.18 || >=5.0.0 <5.0.10 || >=6.0.0 <6.1.9 found in:
 - dependencies: @mojaloop/event-sdk>grpc>@mapbox/node-pre-gyp>tar
```
> Outcome: Fixed

```text
--------------------------------------------------
 yargs-parser needs your attention.

[ low ] Prototype Pollution
 vulnerable versions <13.1.2 || >=14.0.0 <15.0.1 || >=16.0.0 <18.1.2 found in:
 - dependencies: @mojaloop/central-services-shared>widdershins>yargs>yargs-parser
```
> Outcome: Ignored for a week
> Impact: Minimal as the dependencies are used for the Developer Documentation end-point

```text
--------------------------------------------------
 sanitize-html needs your attention.

[ moderate ] Improper Input Validation
 vulnerable versions <2.3.1 found in:
 - dependencies: @mojaloop/central-services-shared>shins>sanitize-html
[ moderate ] Improper Input Validation
 vulnerable versions <2.3.2 found in:
 - dependencies: @mojaloop/central-services-shared>shins>sanitize-html
```
> Outcome: Ignored for a week
> Impact: Minimal as the dependencies are used for the Developer Documentation end-point

### CODE REVIEW

File: package.json [hunk 1]
```diff
@@ -2,7 +2,7 @@
   "name": "quoting-service",
   "description": "Quoting Service hosted by a scheme",
   "license": "Apache-2.0",
-  "version": "12.0.7",
+  "version": "12.0.8",
```
Comment (lewisdaly) [inline package.json:5]: It's time to automate this guy...

File: src/model/quotes.js [hunk 2]
```diff
@@ -164,13 +164,18 @@ class QuotesModel {
       throw ErrorHandler.CreateInternalServerFSPIOPError('Missing quoteRequest', null, fspiopSource)
     }
 
-    await this.db.getParticipant(fspiopSource, LOCAL_ENUM.PAYER_DFSP, quoteRequest.amount.currency, ENUM.Accounts.LedgerAccountType.POSITION)
-    await this.db.getParticipant(fspiopDestination, LOCAL_ENUM.PAYEE_DFSP, quoteRequest.amount.currency, ENUM.Accounts.LedgerAccountType.POSITION)
-
     // Following is the validation to make sure valid fsp's are used in the payload for simple routing mode
     if (envConfig.simpleRoutingMode) {
       await this.db.getParticipant(quoteRequest.payer.partyIdInfo.fspId, LOCAL_ENUM.PAYER_DFSP, quoteRequest.amount.currency, ENUM.Accounts.LedgerAccountType.POSITION)
```
Comment (lewisdaly) [inline src/model/quotes.js:169]: I think this is where the original bug was. We need to check that `quoteRequest.payer.partyIdInfo.fspId` and `quoteRequest.payee.partyIdInfo.fspId` exist before talking to the database.

The trouble was, that this was undefined, and passing that undefined through to knex.

**Review by lewisdaly — CHANGES_REQUESTED** [review]

File: src/model/quotes.js (hunk 2 above)
Comment (mdebarros) [reply src/model/quotes.js]: Thanks @lewisdaly, that was left there by accident.

I was doing some internal testing using the unit tests. It is now removed.

File: package.json (hunk 1 above)
Comment (mdebarros) [reply package.json:5]: So, I added `release` and `github-releases` into the CircleCI config.

Hope it works *holds thumbs*

File: .circleci/config.yml [hunk 3]
```diff
@@ -314,6 +315,72 @@ jobs:
...
+            git checkout ${CIRCLE_BRANCH}
+      - run:
+          name: Configure ssh
+          command: |
+            mkdir -p ~/.ssh
+            ssh-keyscan -p 443 ssh.github.com >> ~/.ssh/known_hosts
+            ssh-keyscan github.com >> ~/.ssh/known_hosts
+      - run:
+          name: Generate changelog and bump package version
+          command: npm run release
```
Comment (lewisdaly) [inline .circleci/config.yml:343]: You need to add this to your `package.json` - e.g. https://github.com/mojaloop/central-ledger/blob/master/package.json#L80

File: package.json (hunk 1 above)
Comment (lewisdaly) [reply package.json:5]: Yeah should do

**Review by lewisdaly — CHANGES_REQUESTED** [review]

**Review by lewisdaly — APPROVED** [review]

**Pull request merged**